go/vt/mysqlctl/mysqld.go

```diff
@@ -1251,6 +1259,20 @@ func (mysqld *Mysqld) applyBinlogFile(binlogFile string, includeGTIDs mysql.GTID
 		args := []string{
 			"--defaults-extra-file=" + cnf,
 		}
+
+		// We disable super_read_only, in case it is in the default MySQL startup
+		// parameters.  We do it blindly, since this will fail on MariaDB, which doesn't
+		// have super_read_only This is safe, since we're restarting MySQL after the restore anyway
+		log.Infof("Restore: disabling super_read_only")
+		if err := mysqld.SetSuperReadOnly(false); err != nil {
+			if strings.Contains(err.Error(), strconv.Itoa(mysql.ERUnknownSystemVariable)) {
+				log.Warningf("Restore: server does not know about super_read_only, continuing anyway...")
+			} else {
+				log.Errorf("Restore: unexpected error while trying to set super_read_only: %v", err)
+				return err
+			}
+		}
```

Comment:
How is this function directly/exclusively tied to restores? If someone runs the equivalent of `mysqldump foo.10 | mysql`, it would be bad if mysqld automatically disabled super_read_only rather than protecting you as it should.